src/map_io/map_object_packet.cc

```diff
@@ -155,6 +155,13 @@ void MapObjectPacket::write(FileSystem& fs, EditorGameBase& egbase, MapObjectSav
 		if (mos.is_object_known(obj))
 			continue;
 
+		if (!is_a(Game, &egbase) &&
+		    (is_a(PlayerImmovable, pobj) || is_a(Worker, pobj) || is_a(Ship, pobj))) {
```

Comment:
Unless we can have tribe objects without an owner here or wares are an issue, `pobj.get_owner() != nullptr` should do the job.